README.md

```diff
@@ -93,6 +93,55 @@ LLVM_INSTALL_UTILS=On
 ```
 to run the regression tests on f18.
 
+### LLVM dependency when building f18 with Fortran IR
+
+If you do not want to build Fortran IR, add `-DLINK_WITH_FIR=Off` to f18 cmake
+command and ignore the rest of this section.
+
+If you intend to build f18 with Fortran IR (`-DLINK_WITH_FIR` On by default),
+you must:
+- build LLVM with the same compiler and options as the one you are using
+to build F18.
+- pass `-DCMAKE_CXX_STANDARD=17 -DLLVM_ENABLE_PROJECTS="mlir"`
+to LLVM cmake command.
+- install LLVM somewhere with `make install` in order to get the required
+AddMLIR cmake file (it is not generated in LLVM build directory).
+
+Installing LLVM from packages is most likely not an option as it will not include
+MLIR and not be built following C++17 standard.
+
+MLIR is under active development and the most recent development version
+may be incompatible. A branch named `f18` is available inside LLVM fork in
+https://github.com/flang-compiler/f18-llvm-project. It contains a version of LLVM
+that is known be compatible to build f18 with FIR.
+
+The fastest way to get set up is to do:
+
+```
+cd where/you/want/to/build/llvm
+git clone --depth=1 -b f18 https://github.com/flang-compiler/f18-llvm-project.git
+mkdir build
+mkdir install
+cd build
+cmake ../f18-llvm-project/llvm -DCMAKE_BUILD_TYPE=Release \
+    -DLLVM_ENABLE_PROJECTS=mlir -DCMAKE_CXX_STANDARD=17 \
+    -DLLVM_INSTALL_UTILS=On \
+    -DCMAKE_INSTALL_PREFIX=../install
+make -j
+make install
```

Comment:
I tried these instructions on my Windows laptop using Windows Subsystem for Linux with Ubuntu installed and using GNU 8.3.0 as my compiler.  The "make -j" command caused my laptop to crash.  Presumably, it ran out of memory.  "make" without the "-j" command worked.  I did not try using an option like "make -j 8".  I also was able to build using Ninja rather than make.  This seemed to work better.  To do this, I just added the option "-G Ninja" at the front of the cmake command.  Specifically, my first line was 
```
cmake -G Ninja ../f18-llvm-project/llvm -DCMAKE_BUILD_TYPE=Release \
```
The command to do the build is then just
```
ninja
```
To do the install, it's
```
ninja install
```
This seemed to work better than using make.  Of course, it requires that you have Ninja installed on your build machine.

Reply:
Thanks for testing the instructions ! I have removed the `-j`, that was a bit wild indeed. The examples are given with make to be portable and understandable by a wide audience. It is also consistent with other example (e.g. in `Build F18`). 
People familiar will builds will know how to replace this with Ninja and to set-up there `-j` according to their machine.